content/documentation/4-batch-developer-guides/2-batch/3-data-flow-simple-task.md

```diff
@@ -1,32 +1,104 @@
 
-In this section we will show how to register task applications with Data Flow, create a Task DSL, and deploy the task to Cloud Foundry, Kubernetes and your local machine.
+In this section, we will demonstrate how to register a Spring Cloud Task application with Data Flow, create a task definiton, and deplot the task to Cloud Foundry, Kubernetes and your local machine.
 
 ## Development
 
-Explaine what we are doing to do development wise.
+For this guide, we will use the [simple task](batch-developer-guides/batch/simple-task) sample Spring Cloud Task application, called `billsetuptask`.
+Follow the instructions to code and build the task if you have not done so already.
+We will register a task applicaton, create a simple task definition, and launch the task using the Data Flow Server.
+The Data Flow Server provides a comprehensive [API](http://docs.spring.io/spring-cloud-dataflow/docs/current/reference/htmlsingle/#api-guide) to perform the necessary steps.
+The Data Flow server includes a Data Flow Dashboard web UI client. In addition there is a [Data Flow Shell](http://docs.spring.io/spring-cloud-dataflow/docs/current/reference/htmlsingle/#shell) CLI, available as separate download.
+The CLI and the UI both expose the complete API functionality.
+Which one to use is a matter of preference, but the UI is quite nice so we will feature it here.
+
+### The Data Flow Dashboard
+
+Assuming Data Flow is [installed](installation/) and running on one of the supported platforms, open your browser at `<data-flow-url>/dashboard`. Here, `<data-flow-url>` depends on the platform. Consult the [installation guide](/installation) to determing the base URL for your installation. If Data Flow is running on your local maching, go to http://localhost:9393/dashboard.
 
 ### Application Registration
 
-how to register apps using the CLI and/or (?) UI
+The Data Flow Dashboard will land on the Application Registration view where we will register the sample task.
+
+![Add an application](images/SCDF-add-applications.png)
+
+#### Application Registration Concepts
+
+Applications in Data Flow are registered as named resources so that they may be referenced when using the Data Flow DSL to configure and compose tasks.
+Registration associates a logical application name and type with a physical resource, given by a URI.
+The URI conforms to a [schema](http://docs.spring.io/spring-cloud-dataflow/docs/current/reference/htmlsingle/#spring-cloud-dataflow-register-stream-apps) and may represent a Maven artifact, a Docker image, or an actual `http(s)` or `file` URL.
+Data Flow defines a some logical application types which indicate its role as a streaming component, a task, or a standalone application.
+Spring Cloud Task applications, as you might guess, are always registered as a `task` type.
+
+##### Maven Artifacts
+
+The URI for a Maven artifact is of the form `maven://<groupId>:<artifactId>[:<extension>[:<classifier>]]:<version>`.
+
+[[note | Application Metadata]]
```

Comment:
Not sure we need this note here.   Unless we cover this in the example, it may add to confusion.   @sabbyanandan Do we need to add a discussion on whitelisting?

Reply:
After asking Sabby we can remove this NOTE.

Reply:
Agree, it may be out of place here.  The maven schema copied from the SCDF reference,  I thought needed some explanation.  Also, the registration view includes metadata resource, so the note intended to clarify these points.  It maybe should go somewhere in the reference guide or in the  `getting started` with tasks section which would walk through this flow with something the prebuilt `timestamp` app.

Reply:
I can add a link to the reference docs on white listing in a note on the Batch/Task app guides.